File: app/src/main/java/com/klim/us_stock/di/symbol_details/SymbolDetailsComponent.java
package com.klim.us_stock.di.symbol_details;

import com.klim.us_stock.ui.windows.symbol_details.SymbolDetailsFragment;

import dagger.Subcomponent;

@Subcomponent(modules = {SymbolDetailsModule.class})
public interface SymbolDetailsComponent {

    public void inject(SymbolDetailsFragment fragment);

}

File: src/presentation/symbol_details/src/main/java/com/klim/symbol_details/di/SymbolDetailsComponent.kt
package com.klim.symbol_details.di

import com.klim.coreUi.BaseFragment
import com.klim.dep_in.ApplicationContextProvider
import com.klim.symbol_details.presentation.SymbolDetailsFragment
import dagger.Component

@Component(
    dependencies = [ApplicationContextProvider::class],
    modules = [SymbolDetailsModule::class, ViewModelsBindModule::class]
)
interface SymbolDetailsComponent {

    fun inject(fragment: SymbolDetailsFragment)

    class Initializer private constructor() {
        companion object {
            fun init(appComponent: ApplicationContextProvider, bf: BaseFragment): SymbolDetailsComponent =
                DaggerSymbolDetailsComponent.builder()
                    .applicationContextProvider(appComponent)
                    .build()
        }
    }

}